### PR TITLE
chore: update `nearcore` crates from `0.17` -> `0.20`

### PR DESCRIPTION
**Pull request by dj8yfo**

### CODE REVIEW

File: .github/workflows/test.yml
Comment (dj8yfo) [inline .github/workflows/test.yml:16]:
```bash

error[E0659]: `opentelemetry` is ambiguous
  --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/near-o11y-0.19.0/src/lib.rs:6:9
   |
6  | pub use opentelemetry::OpenTelemetryLevel;
   |         ^^^^^^^^^^^^^ ambiguous name
   |
   = note: ambiguous because of multiple potential import sources
   = note: `opentelemetry` could refer to a crate passed with `--extern`
   = help: use `::opentelemetry` to refer to this crate unambiguously
note: `opentelemetry` could also refer to the module defined here
  --> /home/jerrick/.cargo/registry/src/index.crates.io-6f17d22bba15001f/near-o11y-0.19.0/src/lib.rs:22:1
   |
22 | mod opentelemetry;
   | ^^^^^^^^^^^^^^^^^^
   = help: use `crate::opentelemetry` to refer to this module unambiguously
                                                                                                                                                                                    
For more information about this error, try `rustc --explain E0659`.
error: could not compile `near-o11y` (lib) due to previous error
```

File: near-sdk/Cargo.toml
Comment (dj8yfo) [inline near-sdk/Cargo.toml:43]: @frol any way to disable `wasmer-runtime-core` build?
```bash
   Compiling near-vm-runner v0.19.0 Running `CARGO=/home/user/.rustup/toolchains/
     stable-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=near_vm_runner CARGO_MANIFEST_DIR=/
     home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/near-vm-runner-0.19.0
     CARGO_PKG_AUTHORS='Near Inc <hello@nearprotocol.com>' CARGO_PKG_DESCRIPTION='This
     crate implements the specification of the interface that Near blockchain exposes to
     the smart contracts.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='MIT OR Apache-2.0'
     CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=near-vm-runner CARGO_PKG_README=README.md
     CARGO_PKG_REPOSITORY='https://github.com/near/nearcore' CARGO_PKG_RUST_VERSION=''
     CARGO_PKG_VERSION=0.19.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=19
     CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/home/user/
     Documents/code/rust_work_repos/near/near-sdk-rs/target/debug/deps:/home/user/.rustup/
     toolchains/stable-x86_64-unknown-linux-gnu/lib:/home/user/.rustup/toolchains/
     stable-x86_64-unknown-linux-gnu/lib:/nix/store/8xgb8phqmfn9h971q7dg369h647i1aa0-zlib-1.3/
     lib' /home/user/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc
     --crate-name near_vm_runner --edition=2021 /home/user/.cargo/registry/src/
     index.crates.io-6f17d22bba15001f/near-vm-runner-0.19.0/src/lib.rs --error-format=json
     --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib
     --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 '--allow=clippy::all'
     --deny=warnings '--deny=clippy::suspicious' '--deny=clippy::redundant_clone'
     '--deny=clippy::len_zero' '--deny=clippy::derivable_impls' '--deny=clippy::correctness'
     '--deny=clippy::clone_on_copy' --cfg 'feature="anyhow"' --cfg 'feature="default"'
     --cfg 'feature="near-vm-compiler"' --cfg 'feature="near-vm-compiler-singlepass"'
     --cfg 'feature="near-vm-engine"' --cfg 'feature="near-vm-types"' --cfg
     'feature="near-vm-vm"' --cfg 'feature="near_vm"' --cfg 'feature="wasmer-compiler"'
     --cfg 'feature="wasmer-compiler-singlepass"' --cfg 'feature="wasmer-engine"'
     --cfg 'feature="wasmer-engine-universal"' --cfg 'feature="wasmer-runtime"' --cfg
     'feature="wasmer-runtime-core"' --cfg 'feature="wasmer-types"' --cfg 'feature="wasmer-vm"'
     --cfg 'feature="wasmer0_vm"' --cfg 'feature="wasmer2_vm"' --cfg 'feature="wasmtime"' --cfg
     'feature="wasmtime_vm"' -C metadata=e481b2a995e931cd -C extra-filename=-e481b2a995e931cd
```

File: .github/workflows/test.yml
Comment (dj8yfo) [inline .github/workflows/test.yml:27]: todo: remove section

File: near-sdk/src/environment/env.rs
Comment (dj8yfo) [inline near-sdk/src/environment/env.rs:82]: todo: correct api for doc snippet

File: near-sdk/src/environment/mock/external.rs
Comment (dj8yfo) [inline near-sdk/src/environment/mock/external.rs:17]: todo: replace with `near_vm_runner::logic::mocks::mock_external::MockedExternal`

File: near-sdk/src/environment/mock/mocked_blockchain.rs
Comment (dj8yfo) [inline near-sdk/src/environment/mock/mocked_blockchain.rs:103]: todo: remove

File: near-sdk/src/environment/mock/mocked_blockchain.rs
Comment (dj8yfo) [inline near-sdk/src/environment/mock/mocked_blockchain.rs:133]: todo: remove

File: near-sdk/src/environment/mock/receipt.rs
Comment (dj8yfo) [inline near-sdk/src/environment/mock/receipt.rs:19]: todo: implement `From<near_vm_runner::logic::mocks::mock_external::MockAction>`

File: near-sdk/src/promise.rs
Comment (dj8yfo) [inline near-sdk/src/promise.rs:640]: todo: add secp conversion

File: near-sdk/src/promise.rs
Comment (dj8yfo) [inline near-sdk/src/promise.rs:671]: todo: move this to `From<near_vm_runner::logic::mocks::mock_external::MockAction>`

File: near-sdk/src/store/unordered_set/mod.rs
Comment (dj8yfo) [inline near-sdk/src/store/unordered_set/mod.rs:47]: todo: correct doc section

File: near-sdk/src/test_utils/mod.rs
Comment (dj8yfo) [inline near-sdk/src/test_utils/mod.rs:31]: todo: correct doc section

File: .github/workflows/test.yml
Comment (dj8yfo) [reply .github/workflows/test.yml:16]: buildkite script has to be corrected  https://buildkite.com/nearprotocol/near-sdk-rs/builds/5218#018d0c88-ae78-49e7-8789-577def4a19da

File: near-sdk/Cargo.toml
Comment (dj8yfo) [reply near-sdk/Cargo.toml]: checked that changes from https://github.com/near/nearcore/pull/10423 alleviate the problem with https://github.com/near/near-sdk-rs/pull/1130/commits/22aedbf10d97d9f3c6470eed6fbd5c4e6ff3b7ad

File: near-sdk/src/environment/mock/receipt.rs
Comment (frol) [inline near-sdk/src/environment/mock/receipt.rs:20]: I am trying to understand if `receipt_index` is actually necessary at this level, and if the types mismatch with the blockchain is justified... @dj8yfo What is your understanding here?

@austinabell Can you shed some light here by any chance?

**Review by frol — APPROVED** [review]
Overall, since it is already in a compilable state, I would merge it and iterate in a separate PR.

File: near-sdk/src/environment/mock/receipt.rs
Comment (austinabell) [reply near-sdk/src/environment/mock/receipt.rs:20]: I can't remember the context, and I don't know about the nearcore updates. I can think about it more another time, but I wouldn't wait on my response, sorry!

**Review by frol — APPROVED** [review]
@dj8yfo Great job! Thanks you for pulling it out. I plan to cut 5.0.0-alpha.2 after merging this one

**Pull request merged**
